pkg/upgrade/upgrade.go

```diff
@@ -49,6 +51,38 @@ const (
 	ZosPackage = "zos.flist"
 )
 
+type ChainVersion struct {
+	SafeToUpgrade bool   `json:"safe_to_upgrade"`
+	Version       string `json:"version"`
+}
+
+func getRolloutConfig(env environment.Environment) (ChainVersion, []uint32, error) {
+	config, err := environment.GetConfig()
+	if err != nil {
+		return ChainVersion{}, nil, errors.Wrap(err, "failed to get network config")
+	}
+
+	manager := substrate.NewManager(env.SubstrateURL...)
+
+	con, err := manager.Substrate()
+	if err != nil {
+		return ChainVersion{}, nil, err
+	}
+
+	v, err := con.GetZosVersion()
+	if err != nil {
+		return ChainVersion{}, nil, errors.Wrap(err, "failed to get zos version from chain")
+	}
+
+	var chainVersion ChainVersion
```

Comment:
we need to handle old style for a while just in case it is not updated on some envs